CartaUtils.py

```diff
@@ -82,6 +95,22 @@ def __init__(self):
         # correct grabbing card not available,
         # on your side, or on opponent's side
         self.correctGrabCardStatus = GrabCardStatus.INVALID
+        self.yourGrabCardSide = GrabbingCardSide.NONE
+        self.oppoGrabCardSide = GrabbingCardSide.NONE
+
+    def reset(self):
+        self.savedStartTime = False
+        self.startTime = 0  # time of the grabbing phase's start
+        self.yourGrabCardLastWord = ""
```

Comment:
remove `self.GrabCardLastWord`
remove `self.oppoGrabbedCard`
remove `self.youGrabbedCard`
remove `self.ended`

and modify to the new structures